packages/server/graphql/private/mutations/generateMeetingSummary.ts

```diff
@@ -0,0 +1,185 @@
+      .load(meetingId)
+    const reflectionGroups = Promise.all(
+      rawReflectionGroups
+        .filter((g) => g.voterIds.length > 1)
+        .map(async (group) => {
+          const {id: reflectionGroupId, voterIds, title} = group
+          const [comments, rawReflections, discussion] = await Promise.all([
+            getComments(reflectionGroupId),
+            dataLoader.get('retroReflectionsByGroupId').load(group.id),
+            pg
+              .selectFrom('Discussion')
+              .selectAll()
+              .where('discussionTopicId', '=', reflectionGroupId)
+              .limit(1)
+              .executeTakeFirst()
+          ])
+          const discussPhase = getPhase(meeting.phases, 'discuss')
+          const {stages} = discussPhase
+          const stageIdx = stages
+            .sort((a, b) => (a.sortOrder < b.sortOrder ? -1 : 1))
+            .findIndex((stage) => stage.discussionId === discussion?.id)
+          const discussionIdx = stageIdx + 1
+
+          const reflections = await Promise.all(
+            rawReflections.map(async (reflection) => {
+              const {promptId, creatorId, plaintextContent} = reflection
+              const [prompt, creator] = await Promise.all([
+                dataLoader.get('reflectPrompts').load(promptId),
+                creatorId ? dataLoader.get('users').loadNonNull(creatorId) : null
+              ])
+              const {question} = prompt
+              const creatorName = disableAnonymity && creator ? creator.preferredName : 'Anonymous'
+              return {
+                prompt: question,
+                author: creatorName,
+                text: plaintextContent,
+                discussionId: discussionIdx
+              }
+            })
+          )
+          const shortMeetingDate = new Date(meetingDate).toISOString().split('T')[0]
+          const res = {
+            voteCount: voterIds.length,
+            title: title,
+            comments,
+            reflections,
+            meetingName,
+            date: shortMeetingDate,
+            meetingId,
+            discussionId: discussionIdx
+          }
+
+          if (!res.comments || !res.comments.length) {
+            delete (res as any).comments
+          }
+          return res
+        })
+    )
+
+    return reflectionGroups
+  }
+  const manager = new OpenAIServerManager()
+
+  const updatedMeetingIds = await Promise.all(
+    rawMeetings.map(async (meeting) => {
+      const meetingsContent = await getMeetingsContent(meeting)
+      if (!meetingsContent || meetingsContent.length === 0) {
+        return null
+      }
+      const yamlData = yaml.dump(meetingsContent, {
+        noCompatMode: true
+      })
+      const newSummary = await manager.generateSummary(yamlData, prompt as string)
+      if (!newSummary) return null
+
+      const now = new Date()
+      await r
+        .table('NewMeeting')
+        .get(meeting.id)
+        .update({summary: newSummary, updatedAt: now})
```

Comment:
+1 overwriting a user-facing value on an ad-hoc basis feels a little dirty. It means the experiments we run will be immediately visible to the user. Probably OK for now if we assume the prompt isn't going to change!